src/latent_gp.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    LatentGP(f<:GP, x<:AbstractVector, Φ)
+
+ - `f` is a Gaussian Process.
+ - `x` is an `AbstractVector` of inputs of type `Tx`.
+ - `Φ` is the log likelihood function which maps input of type `Tx` to `Real`.
+    
+"""
+struct LatentGP{T<:GP,Tx}
+    f::T
+    x::AbstractVector{Tx}
+    σ²
```

Comment:
I'm don't think that we need a noise variance parameter for this model, do we? This should be handled by the likelihood now.

edit: this is superseded by my next comment. In general you'll want the noise variance in `fx` to be something close to zero, but typically it won't quite be zero for stability reasons. Either way, we shouldn't have to worry about this too much in this package I don't think, at least not at the API level.

Reply:
Also, I think that it makes sense to just keep a single `FiniteGP` object `fx` rather than separately storing `f` and `x`.

Reply:
If we were to keep a single `FiniteGP` object, it would incorporate the noise too. This should solve both the issues.